In [1]:
import pandas as pd
import numpy as np
from pandas_profiling import ProfileReport

In [2]:
def read_data():
    print('Reading files...')    
    order_df = pd.read_csv('../input/machine_learning_challenge_order_data.csv')
    print('Order data has {} rows and {} columns'.format(order_df.shape[0], order_df.shape[1]))
    label_df = pd.read_csv('../input/machine_learning_challenge_labeled_data.csv')
    print('Label data has {} rows and {} columns'.format(label_df.shape[0], label_df.shape[1]))
    df = order_df.merge(label_df, on='customer_id')
    return df

df = read_data()

Reading files...
Order data has 786600 rows and 13 columns
Label data has 245455 rows and 2 columns


In [ ]:
profile = ProfileReport(df)
profile.to_file(output_file='output.html')
profile

In [ ]:
profile.to_widgets()

***Warnings by pandas-profiling***
- Dataset has 546 (0.1%) duplicate rows
- customer_id has a high cardinality: 245455 distinct values
- order_date has a high cardinality: 776 distinct values
- customer_order_rank has 24767 (3.1%) missing values
- voucher_amount is highly skewed (γ1 = 30.39394065)
- platform_id is highly skewed (γ1 = -22.53663783)
- voucher_amount has 743462 (94.5%) zeros
- delivery_fee has 597536 (76.0%) zeros